### DISCUSSION TITLE
Flows: Terminate a flow on fail condition? and transform payload... before insert/update?

### DISCUSSION THREAD

**Discussion by ddelyon** [directus/directus#15019] [Archive]
Great to see the improvements in Flows, thanks one and all, so good!
Two hopefully easy to answer queries that I just have not been able to work out, read about, or find a solution to - hoping someone can spare a minute or two to help or advise...

1. With an event hook, filter (blocking),  on item.create AND item.update - I have a condition operation, if true, flow continues to next operation... if false I want the whole flow to stop,
and the create or update to not proceed.
Is this possible with current functionality? (9.15.1)

2. Again, with an event hook, filter(blocking), I want to ADD a field name/value to the payload on item.create or update. I had hoped transform operation would enable me to do this.
That does not appear to be the case nor the intention of the transform payload. Can it be done with existing functionality. (9.15.1) ?

Realise everyone is super busy. Any help, really appreciated :-)

**Comment by azrikahar** [comment]
> 1. With an event hook, filter (blocking),  on item.create AND item.update - I have a condition operation, if true, flow continues to next operation... if false I want the whole flow to stop,
>    and the create or update to not proceed.
>    Is this possible with current functionality? (9.15.1)

It is currently not possible to stop it, but #15283 aims to add this exact interaction 👍

> 2. Again, with an event hook, filter(blocking), I want to ADD a field name/value to the payload on item.create or update. I had hoped transform operation would enable me to do this.
>    That does not appear to be the case nor the intention of the transform payload. Can it be done with existing functionality. (9.15.1) ?

I'm not 100% sure about this myself, but my current guess on why it may not currently work is Transform operation _should_ work to ADD the new field/value, but likely the _result_ of the Transform operation isn't being return as the input, hence it's as if it was never transformed, in a sense. Would be a good use case to eventually support nonetheless!

**Reply by ddelyon** [reply]
Hi Azri, thanks so much for taking the time to reply. I've been really struggling to get my head around these two queries!
1. Really hope that #15283 or similar functionality appears. Having the ability to stop a CRUD  action within a (blocking) filter flow would be SO useful. While data validation at input is really flexible, there are so many use cases to cause a requested 'save' to be blocked. Filter(blocking) being able to block would do it.

2. My use case here is simple... I want to be able to create a 'computed' field value prior to create/update. 
I really hoped a filter (blocking) flow would enable me to do this, avoiding the need to create/update the record first, then have a more complicated update approach by, for example triggering another flow.
I might be missing something, but not being able to manipulate the payload of a filter based flow, negates some of the benefits of the blocking concept.
I've been trying to get somewhere with operations and script operations, but no luck yet. Hopefully some draft docs/cookbook that has been mentioned by the team will help us all get a bit more up to speed with the thinking behind it. I am sure that is all it is - I don't know all I need to know !
Cheers and thanks again!
